drivers/hmis/app/graphql/concerns/graphql_permission_checker.rb

```diff
@@ -22,7 +18,20 @@ def self.current_permission_for_context?(context, permission:, entity:)
         context.dataloader.with(Sources::ActiveRecordAssociation, association).load(record)
       end
     end
-    raise "Missing loader for #{entity.class.name}##{entity.id}" unless loader
+    raise "Missing loader for #{entity.class.name}##{entity.id}" unless loader && subject
+
+    # subject is expected to be one of Client, Project, Organization, or Data Source.
+    data_source_id = if subject.is_a? GrdaWarehouse::DataSource
+      subject.id
+    else
+      subject.data_source_id
+    end
+
+    # Return false because even if the user has permission to view this record in another hmis data source,
+    # they should not be able to resolve it in the context of the hmis they are currently using (hmis_data_source_id).
+    # The viewable_by scope is our primary defense against this; this object-level authorization is a secondary guard.
+    # But caution because it's not applied 100% consistently; for example, organizations don't have object-level authorization
```

Comment:
So this doesn't feel like the totally right place for this warning, since this class `GraphqlPermissionChecker` _happens_ to be used for object-level authorization, but its not ONLY used for object-level authorization. Maybe you could put this comment on the `self.authorized?` implementations?


It would also be great to have this in the README. Would you mind adding an "faq" or section about this? Here is a rough description of what we discussed earlier:

**Authorization considerations when writing GraphQL queries and mutations**
* Most HMIS records have a scope called `viewable_by`. Always use this scope when looking up a record. Most HMIS records' viewable_by scopes also perform a filter on `current_user.hmis_data_source_id`, ensuring that objects from other HMIS data sources are not included.
* When checking permissions, 90% of the time we want to check against a specific entity. (Eg can the user edit THIS enrollment, not just can the user edit any enrollments). The preferred way to do this is using `current_permission?(permission:, entity:)` because it uses the graphql data loader and also ensures that the entity belongs to the correct data source.
* For nodes that are resolved from lots of places in the graph (and have PII), add object-level authorization to the object type (`def self.authorized?`). These are a secondary check to ensure the presence of the minimal permissions needed to resolve the object.